### PR TITLE
Frontend PanelChrome: Fix header title moving down when collapsing

### LINKED ISSUE #74120: Frontend PanelChrome: Title shifts 1px down when collapsing
Title shifts 1px down when collapsing PanelChrome. 

https://github.com/grafana/grafana/assets/58232930/1f009761-a830-46b2-a861-e6d240e4ab86

### PR DESCRIPTION
**Pull request by harisrozajac**
Fixed PanelChrome's header title moving down 1px when collapsing.

@torkelo also fixed: 
- Icon placement (alignItems: center did not work as the button did not have display: flex), and corrected the spacing for the icon with a gap property.
- Also a minor refactoring moved the collapsing header into where it's rendered and simplified the conditional logic there

Before:

https://github.com/grafana/grafana/assets/58232930/bac3d925-b56e-49df-b3ea-f2af568ab9bf

After:

https://github.com/grafana/grafana/assets/58232930/5c5e7973-3383-4234-b71d-cc054f3ab9b3

Fixes: https://github.com/grafana/grafana/issues/74120

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 1]
```diff
@@ -351,7 +351,7 @@ const getStyles = (theme: GrafanaTheme2) => {
     container: css({
       label: 'panel-container',
       backgroundColor: background,
-      border: `1px solid ${borderColor}`,
+      outline: `1px solid ${borderColor}`,
```
[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:354]: not sure how the border impacts the placement of the title in the header feels like this requires a fix in the PanelChrome header?

[the remark this answers is not in the document]
Comment (harisrozajac) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx]: @torkelo ah that makes sense. After looking into this a bit more, I've found that there was discrepancy in container height when collapsing it. Previously, when header was collapsed the height was set to `headerHeight`, when the border should have been accounted for also, so I've added `panelBorder` to that calculation.

[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx]: @harisrozajac I would fix it by simply not setting height on the container when in collapse state (there already is a height on the header). 

I have a branch here where I tried that:
https://github.com/grafana/grafana/compare/fix-panel-chrome-collapse-pos  

I also fixed
* Icon placement (alignItems: center did not work as the button did not have display: flex), and corrected the spacing for the icon with a gap property. 
* Also a minor refactoring moved the collapsing header into where it's rendered and simplified the conditional logic there (I try not to use the conditional (ternary) operator in rendering if can avoid it).

[the remark this answers is not in the document]
Comment (harisrozajac) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx]: @torkelo i've made the update for the header when collapsing. should we add the other changes you made on your branch to this PR, or maybe it's better to have them in a separate one?

[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx]: @harisrozajac if you can add them to this PR that be great

[the remark this answers is not in the document]
Comment (harisrozajac) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx]: sure thing!

**Review by torkelo — APPROVED** [review]

**Pull request merged**